app/code/Bulbulatory/Recommendations/Ui/Component/Listing/Column/RecommenderEmail.php

```diff
@@ -0,0 +1,39 @@
+<?php
+namespace Bulbulatory\Recommendations\Ui\Component\Listing\Column;
+
+use Magento\Framework\View\Element\UiComponentFactory;
+use Magento\Framework\View\Element\UiComponent\ContextInterface;
+use Magento\Ui\Component\Listing\Columns\Column;
+use \Magento\Customer\Api\CustomerRepositoryInterface;
+
+class RecommenderEmail extends Column
+{
+    protected $customerRepositoryInterface;
+
+    public function __construct(
+        ContextInterface $context,
+        UiComponentFactory $uiComponentFactory,
+        CustomerRepositoryInterface $customerRepositoryInterface,
+        array $components = [],
+        array $data = []
+    ) {
+        parent::__construct($context, $uiComponentFactory, $components, $data);
+        $this->_customerRepositoryInterface = $customerRepositoryInterface;
+    }
+
+public function prepareDataSource(array $dataSource)
+{
+    if (isset($dataSource['data']['items'])) {
+        foreach ($dataSource['data']['items'] as &$item) {
+            if ($item['customer_id']) {
+                $recommenderId = $item['customer_id'];
+                $recommender = $this->_customerRepositoryInterface->getById($recommenderId);
```

Comment:
Tutaj możesz wstawić bezpośrednio `$item['customer_id']` jako parametr - wtedy linia wyżej będzie niepotrzebna:

```php
$recommender = $this->_customerRepositoryInterface->getById($item['customer_id']);
```

Reply:
Tutaj zasugerowałbym również pewną optymalizację. W tej chwili dla każdego wyświetlanego wiersza pobierasz obiekt aby zdobyć jego adres e-mail. To daje liczbę operacji równą liczbie wierszy. Zastanów się czy nie lepiej najpierw zebrać wszystkie `customer_id` a potem zapytać o listę customerów 1 raz a następnie uzupełniać grida nimi. To nie jest konieczne i możesz to traktować jako zadanie z gwiazdką :P